pytorch_lightning/utilities/imports.py

```diff
@@ -94,6 +94,7 @@ def _compare_version(package: str, op: Callable, version: str, use_base_version:
 _RICH_AVAILABLE = _module_available("rich") and _compare_version("rich", operator.ge, "10.2.2")
 _TORCH_QUANTIZE_AVAILABLE = bool([eg for eg in torch.backends.quantized.supported_engines if eg != "none"])
 _TORCHTEXT_AVAILABLE = _module_available("torchtext")
+_TORCHTEXT_LEGACY_DEPRECATED: bool = _compare_version("torchtext", operator.gt, "0.11.0")
```

Comment:
should the goal of this PR be that Lightning will warn users that have a version greater than 0.11? 
If we deprecate Lightning's support for the legacy torchtext Batch, we would want to print a message to the users, right? i.e., deprecate now in v1.6 and remove support in 1.8?

Reply:
can we have instead

```suggestion
_TORCHTEXT_LEGACY = _TORCHTEXT_AVAILABLE and _compare_version("torchtext", operator.gt, "0.11.0")
```

Also, why is the version 0.11 and not 0.9? isn't 0.9 where it was deprecated?

Reply:
@carmocca I agree with your suggestion. Also the version is 0.11 because we (torchtext) remove it in 0.11. 0.9 was when we migrated it tot `torchtext.legacy`.

@awaelchli Regarding your comment, I am not sure what is the best practice in Lightning to deprecate items. Either of these work fine for me. What is your recommendation. If you want to show a warning, what is the best place/way to do that ?

Reply:
I left a comment where I think the warning should be displayed https://github.com/PyTorchLightning/pytorch-lightning/pull/10724#discussion_r756469670